### ISSUE TITLE
Pre-defined Microsoft Teams webhook fails because of broken notification messaging templates for some languages

### ISSUE THREAD

**Issue by afloquet** [zammad/zammad#5410]
### Used Zammad Version

6.4.0-4

### Environment

- Installation method: deployment with [Helm template](https://artifacthub.io/packages/helm/zammad/zammad/13.0.1) on OpenShift cluster
- Operating system (if you're unsure: `cat /etc/os-release` ): containers based on Debian 12
- Database + version:  postgresql 17.0
- Elasticsearch version: 8.15.3
- Browser + version: Microsoft Edge 130.0.2849.80
- Language used: French

### Actual behaviour

This is quite similar to the previous issue I had previously opened #5218, but it was never solved for us and is still present today even after updates and trying to check any template for the source of this error (but couldn't get any precise stacktrace of what is happening behind)

When configuring a predefined webhook for Microsoft Teams notification or trying to test with custom payload or fully custom webhook, the notification is not sent and the following log can be found in the scheduler (sensitive information has been replaced) :

```
# Update of an existing ticket
zammad-scheduler E, [2024-11-12T18:12:01.148433#1-180040] ERROR -- : 2024-11-12T18:12:01+0000: [Worker(host:zammad-scheduler-7c4ffd7dc8-swwc4 pid:1)] Job TriggerWebhookJob [a7b5abba-7f73-44bb-87cd-2b1ba7b86f5d] from DelayedJob(default) with arguments: [{"_aj_globalid"=>"gid://zammad/Trigger/10"}, {"_aj_globalid"=>"gid://zammad/Ticket/5587"}, nil, {"changes"=>{"Owner"=>["<REPLACED: Name of previous owner>", "<REPLACED: Name of new owner>"], "_aj_symbol_keys"=>[]}, "user_id"=>5, "execution_type"=>"trigger", "event_type"=>"update", "_aj_ruby2_keywords"=>["changes", "user_id", "execution_type", "event_type"]}] (id=2546537) (queue=default) FAILED (0 prior attempts) with NoMethodError: undefined method `[]' for nil:NilClass
zammad-scheduler E, [2024-11-12T18:12:01.148527#1-180040] ERROR -- : 2024-11-12T18:12:01+0000: [Worker(host:zammad-scheduler-7c4ffd7dc8-swwc4 pid:1)] Job TriggerWebhookJob [a7b5abba-7f73-44bb-87cd-2b1ba7b86f5d] from DelayedJob(default) with arguments: [{"_aj_globalid"=>"gid://zammad/Trigger/10"}, {"_aj_globalid"=>"gid://zammad/Ticket/5587"}, nil, {"changes"=>{"Owner"=>["<REPLACED: Name of previous owner>", "<REPLACED: Name of new owner>"], "_aj_symbol_keys"=>[]}, "user_id"=>5, "execution_type"=>"trigger", "event_type"=>"update", "_aj_ruby2_keywords"=>["changes", "user_id", "execution_type", "event_type"]}] (id=2546537) (queue=default) FAILED permanently because of 1 consecutive failures

# Creating a new ticket
zammad-scheduler E, [2024-11-12T18:52:49.043603#1-180040] ERROR -- : 2024-11-12T18:52:49+0000: [Worker(host:zammad-scheduler-7c4ffd7dc8-swwc4 pid:1)] Job TriggerWebhookJob [b85453c8-3b84-42ae-9941-7307e5ed10a0] from DelayedJob(default) with arguments: [{"_aj_globalid"=>"gid://zammad/Trigger/10"}, {"_aj_globalid"=>"gid://zammad/Ticket/5890"}, {"_aj_globalid"=>"gid://zammad/Ticket::Article/24907"}, {"changes"=>{"_aj_symbol_keys"=>[]}, "user_id"=>5, "execution_type"=>"trigger", "event_type"=>"create", "_aj_ruby2_keywords"=>["changes", "user_id", "execution_type", "event_type"]}] (id=2546556) (queue=default) FAILED (0 prior attempts) with NoMethodError: undefined method `[]' for nil:NilClass
zammad-scheduler E, [2024-11-12T18:52:49.044452#1-180040] ERROR -- : 2024-11-12T18:52:49+0000: [Worker(host:zammad-scheduler-7c4ffd7dc8-swwc4 pid:1)] Job TriggerWebhookJob [b85453c8-3b84-42ae-9941-7307e5ed10a0] from DelayedJob(default) with arguments: [{"_aj_globalid"=>"gid://zammad/Trigger/10"}, {"_aj_globalid"=>"gid://zammad/Ticket/5890"}, {"_aj_globalid"=>"gid://zammad/Ticket::Article/24907"}, {"changes"=>{"_aj_symbol_keys"=>[]}, "user_id"=>5, "execution_type"=>"trigger", "event_type"=>"create", "_aj_ruby2_keywords"=>["changes", "user_id", "execution_type", "event_type"]}] (id=2546556) (queue=default) FAILED permanently because of 1 consecutive failures 
```

We can see that it got triggered but failed with undefined method `[]' for nil:NilClass. No log is present in the Webhook section of Zammad settings

The webhook as been defined as follow :
![Image](https://github.com/user-attachments/assets/b60dad69-d161-456b-b53c-85fd1645b89b)

The trigger has been defined as follow :
![Image](https://github.com/user-attachments/assets/2ceb75b9-418a-4a85-ad43-9b4a2bf0d1a4)

When testing outside of Zammad with Postman or equivalent with an equivalent payload, the notification is sent with success and received in Teams.

The install has been upgraded to latest version on regular basis since December 2021 and is fully up to date as of today. No change has been made to any code out of the updates. Webhooks used to work until december 2023 in our case.

Your help would be much appreciated

### Expected behaviour

The webhook is triggered, which is reachable in the webhook logs in Zammad and received in Microsoft Teams

### Steps to reproduce the behaviour

1. Zammad deployed with Helm on Openshift and regularly updated with no custom change
1. Configure a predefined webhook with the URL of an Incoming Webhook connector on Microsoft Teams
1. Configure a trigger to track any open or closed ticket, or for which agent changes
1. Try to change the agent on some ticket / Try to create a new ticket
1. Notification is not sent as described above

### I'm sure this is a bug and no feature request or a general question.

yes

**Comment by fliebe92** [comment]
Hi @afloquet. Could you please provide more logs? Without a traceback, it's almost impossible to know the location in the code where it fails for you. By the way, it works fine for me...

**Comment by YannC-vialtic** [comment]
Hello
Same issue for me on 6.3.1-1728990975.00dde803.bullseye version.

**Comment by MrGeneration** [comment]
> 6.3.1

Zammad 6.3 is outdated. The pre defined webhook payloads changed for Microsoft Teams in between those versions and thus your "me too" might be of different cause.

**Comment by afloquet** [comment]
Hi @fliebe92,

I don't have any traceback at the moment, only the log messages sent in the first comment.
I'd be happy to provide more logs, I'm trying to increase log level but couldn't find any related value in the Helm template.
Also, the filesystem ends up read-only so I cannot change anything in the pod as is..

Maybe you have some advice to get those additional logs ?
I'm still looking in the meantime and will update here if I find any useful info

**Comment by zagub59** [comment]
Hi there, 

Just to add up that I have the same issue with predefined Slack and Rocket.chat webhooks on latest Zammad version (6.4.0-1731502166.7fd61b3b.bookworm).

Just opened an issue on the Technical Assistance forum : https://community.zammad.org/t/webhooks-not-working/16014/2

EDIT : just tried setting up the system in english instead of French and webhooks are now working so it has something to do with French language :)

**Comment by fliebe92** [comment]
Someone broke the functionality by editing the French templates.

**Comment by afloquet** [comment]
Hello,

I can confirm that the issue is solved in the [version 13.0.2 of the helm template ](https://artifacthub.io/packages/helm/zammad/zammad/13.0.2) just deployed now

Everything works as it should again 😄 

Thank you for your help and support !
